### DISCUSSION TITLE
preview of type renaming in VSCode (WIP)

### DISCUSSION THREAD

**Discussion by AlexHaxe** [HaxeCheckstyle/haxe-rename#1] [Show and tell]
![20201209_rename_type_wip](https://user-images.githubusercontent.com/7204032/101621414-01dec400-3a16-11eb-8365-fc518b6ce1c6.gif)

**Comment by AlexHaxe** [comment]
how to build and install your own language server:
```bash
$ gh repo clone vshaxe/haxe-language-server
$ cd haxe-language-server/
$ git checkout rename
$ npm install
$ npx lix install gh:HaxeCheckstyle/haxe-rename
$ npx lix run vshaxe-build -t language-server --debug
$ cp ~/.vscode/extensions/nadako.vshaxe-2.21.4/server/bin/server.js ~/.vscode/extensions/nadako.vshaxe-2.21.4/server/bin/server.bak.js
$ cp bin/server.js ~/.vscode/extensions/nadako.vshaxe-2.21.4/server/bin/
```
for Windows: replace `~` with `%HOMEPATH%` (and `cp` with `copy`)

restart languageserver / vscode to make it load and use new server.js

only works on projects with Haxe sources in `src` folder (hardcoded location in languageserver)

**Reply by romamik** [reply]
This should be in formatter readme!
